src/utils/DMRoomMap.js

```diff
@@ -94,10 +97,13 @@ export default class DMRoomMap {
      * with ourself, not the other user. Fix it by guessing the other user and
      * modifying userToRooms
      */
-    _patchUpSelfDMs(userToRooms) {
+    async _patchUpSelfDMs(userToRooms) {
         const myUserId = this.matrixClient.getUserId();
         const selfRoomIds = userToRooms[myUserId];
         if (selfRoomIds) {
+            // account data gets emitted before the rooms are available
+            // so wait for the sync to be ready and then read the rooms.
+            await this._waitForSyncReady();
```

Comment:
`populateRoomToUser()` is done on access if necessary - could we do the same here, on the assumption that everything will be ready by the time anything comes to query the DMRoomMap? Introducing async-ness to wait for the sync feels like adding more moving parts to go wrong and could introduce races where the DMRoomMap hasn't yet been updated.

Reply:
Yeah, that's a better idea indeed.